src/js/pouch.js

```diff
@@ -25,42 +46,79 @@ gpii.pouch.init = function (that) {
     fs.writeFileSync(that.options.pouchConfigPath, JSON.stringify(that.options.pouchConfig, null, 2));
 
     var MemPouchDB = PouchDB.defaults({ db: memdown });
-
-    fluid.each(that.options.databases, function (dbConfig, key) {
-        var db = new MemPouchDB(key);
-        if (dbConfig.data) {
-            var data = require(dbConfig.data);
-            db.bulkDocs(data);
-        }
-    });
-
     that.expressPouchdb = expressPouchdb(MemPouchDB, { configPath: pouchConfigPath });
 
-    //that.expressPouchdb.couchLogger.setFile(logPath);
+    if (PouchDB.isBeingCleaned) {
+        fluid.fail("I should never be allowed to load data if pouch is still being cleaned from the previous run...");
+    }
+    else {
+        var promises = [];
+        fluid.each(that.options.databases, function (dbConfig, key) {
+            var db = new MemPouchDB(key);
+            that.databaseInstances[key] = db;
+            if (dbConfig.data) {
+                var data = require(dbConfig.data);
+                promises.push(db.bulkDocs(data));
+            }
+        });
 
-    that.events.onStarted.fire();
+        when.all(promises).then(function () {
+            that.events.onStarted.fire();
+        });
+    }
 };
 
 gpii.pouch.getRouter = function (that) {
     return that.expressPouchdb;
 };
 
+// Remove all data from each database between runs, otherwise we encounter problems with data leaking between tests.
+//
+// https://github.com/pouchdb/pouchdb/issues/4124
+//
+gpii.pouch.cleanup = function (that) {
+    fluid.each(that.databaseInstances, function (db, key) {
+            db
+                .allDocs()
+                .then(function (result) {
+                    var bulkPayloadDocs = fluid.transform(result.rows, gpii.pouch.transformRecord);
+                    var bulkPayload     = {docs: bulkPayloadDocs};
+                    return db.bulkDocs(bulkPayload);
+                })
+                .then(function () {
+                    fluid.log("Deleted existing data from database '" + key + "'...");
+                    return db.compact();
+                })
+                .then(function (result) {
+                    fluid.log("Compacted existing database '" + key + "'...");
+                })
+                .catch(fluid.fail); // jshint ignore:line
 
-// TODO:  Write a change listener to allow easy adding of new databases
+        }
+    );
+};
 
-/*
-    The "databases" option is expected to be an array keyed by dbName, with options to control whether data is loaded or not, as in:
+gpii.pouch.transformRecord = function (record) {
+    // We cannot use "that" or its options here because we have already been destroyed by the time this function is called.
```

Comment:
So, the following transcript (with a few extra messages) indicates that your current implementation is just working through sheer luck - the extra delay of starting the express server on the second run is probably enough to delay it just long enough that the database has been cleaned:

12:27:57.837:  Successfully queued test Testing initial pouch load...
12:27:57.842:  Successfully queued test Testing second pouch load...
12:27:58.356:  Express server listening on port 6792
12:27:58.359:  Express started...
12:27:58.379:  Sending a GET request to: /sample/ on port 6792
jq:  Test concluded - Test name "Testing initial pouch load...": 1/1 passed - PASS
12:27:58.982:  Destroying pouchDB database sample
12:27:59.008:  Express stopped...
12:27:59.013:  Express server listening on port 6792
12:27:59.016:  Express started...
12:27:59.022:  Beginning to scan database data for destruction
12:27:59.031:  Deleted existing data from database 'sample'...
12:27:59.042:  Sending a GET request to: /sample/ on port 6792
12:27:59.062:  Compacted existing database 'sample'...
12:27:59.076:  Destroying pouchDB database sample
12:27:59.079:  Express stopped...
jq:  Test concluded - Test name "Testing second pouch load...": 1/1 passed - PASS
12:27:59.090:  Beginning to scan database data for destruction
jq:  ***************
jq:  All tests concluded: 2/2 total passed in 2329ms - PASS
jq:  ***************

See comment above for suggestion about making this work properly